### PR TITLE
Implemented RestartUnit and ReloadUnit

### PR DESCRIPTION
**Pull request by engelmi**
This PR implements the `RestartUnit` and `ReloadUnit` interface for the manager and the agent.

### CODE REVIEW

File: src/agent/agent.c [hunk 1]
```diff
@@ -388,87 +386,52 @@ static int agent_method_start_unit(UNUSED sd_bus_message *m, void *userdata, UNU
...
-                            job_object_path,
-                            agent_job_done,
-                            agent_job_op_ref(op),
-                            (free_func_t) agent_job_op_unref)) {
-                return sd_bus_reply_method_errorf(req->request_message, SD_BUS_ERROR_FAILED, "Internal Error");
-        }
-
-        return sd_bus_reply_method_return(req->request_message, "");
+static int agent_method_start_unit(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        return agent_run_unit_lifecycle_method(m, (Agent *) userdata, "StartUnit");
```
Comment (engelmi) [inline src/agent/agent.c:401]: The code for `StartUnit` and `StopUnit` is mostly the same - the same applies to the new `RestartUnit` and `ReloadUnit`. Therefore, I extracted it into a dedicated function. Not sure about the naming, though.

File: src/manager/node.c [hunk 2]
```diff
@@ -570,61 +569,36 @@ static int node_method_start_unit(sd_bus_message *m, void *userdata, UNUSED sd_b
...
+/*************************************************************************
+ ********** org.containers.hirte.Node.RestartUnit ************************
+ ************************************************************************/
 
-        r = sd_bus_message_append(req->message, "ssu", unit, mode, setup->job->id);
-        if (r < 0) {
-                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
-        }
+static int node_method_restart_unit(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        return node_run_unit_lifecycle_method(m, (Node *) userdata, "Restart", "RestartUnit");
```
Comment (alexlarsson) [inline src/manager/node.c]: Actually, i looked at what systemd does and the "jobtype" is:
> JobType refers to the job's type and is one of start, verify-active, stop, reload, restart, try-restart, reload-or-start. Note that later versions might define additional values. 

So, no capital letters in the beginning.

File: src/agent/agent.c (hunk 1 above)
Comment (alexlarsson) [reply src/agent/agent.c:401]: sounds fine

File: src/manager/node.c (hunk 2 above)
Comment (engelmi) [reply src/manager/node.c]: Ah ok, I aligned it accordingly.

**Review by alexlarsson — APPROVED** [review]

**Pull request merged**